_health-care/_js/reducers/veteran/index.js

```diff
@@ -199,6 +206,20 @@ function veteran(state = blankVeteran, action) {
       Object.assign(pathToData(newState, action.path), initializeNullValues(pathToData(newState, action.path)));
       return newState;
 
+    case ENSURE_CHILD_FIELDS_INITIALIZED:
+      newState = Object.assign({}, state);
+      // update children income from children info
+      newState.annualIncome.children.splice(newState.childInformation.children.length);
+      for (let i = 0; i < newState.childInformation.children.length; i++) {
+        const shortName = `${newState.childInformation.children[i].childFullName.first} ${newState.childInformation.children[i].childFullName.last}`;
```

Comment:
The convention in React is not to store computed data in state. Since `shortName` is computed, we can probably find a way to move this out of our action and instead to `ChildIncome.jsx`.